src/Toast.js

```diff
@@ -69,17 +69,28 @@ const Toast = ({
   innerRef,
   ...props
 }) => {
+  const delayRef = useRef(delay);
+  const onCloseRef = useRef(onClose);
+
+  // We use refs for these, because we don't want to restart the autohide
+  // timer in case these values change.
+  delayRef.current = delay;
+  onCloseRef.current = onClose;
+
   useEffect(() => {
-    if (autohide && show) {
-      const timer = setTimeout(() => {
-        onClose();
-      }, delay);
-      return () => {
-        clearTimeout(timer);
-      };
+    if (!(autohide && show)) {
+      return undefined;
     }
-    return () => null;
-  }, [autohide, show, delay, onClose]);
+
+    const autohideHandle = setTimeout(() => {
+      onCloseRef.current();
+    }, delayRef.current);
+
+    return () => {
+      clearTimeout(autohideHandle);
+    };
+  }, [autohide, show]);
```

Comment:
We don't actually want to restart the timer if `delay` or `onClose` change on us, I think.